app/controllers/vendors_controller.rb

```diff
@@ -1,7 +1,11 @@
 class VendorsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :authorize_user!, only: [:update, :destory]
```

Comment:
So as of right now, once a normal user makes a submission of a vendor, they won't ever be able to edit it unless they become an admin. Is this your intention?